### DISCUSSION TITLE
book, ¿training, fine-tuning, large model?

### DISCUSSION THREAD

**Discussion by SilvaRaulEnrique** [ggml-org/llama.cpp#3439] [Q&A]
what is best, and what is the difference? training, fine-tuning or use large model.
I want llama read a book and make answered to my questions.
¿what order to use?

**Comment by staviq** [comment]
> what is best 

Neither probably, assuming what you mean is that you want to "give" the AI a book and make it "read" it.

You can try with [private gpt](https://github.com/imartinez/privateGPT), it's designed exactly for that.

But if you want to implement that yourself, with llama.cpp, this is typically done with vector stores, embedding queries, and agents, which llama.cpo could do, but you'd have to write your own code for the agent, as this isn't directly provided by llama.cpp.

**Comment by SilvaRaulEnrique** [comment]
Thankyou.
But, for llama.cpp to read a book in .txt, it must be trained, or fine-tuning is sufficient, or a long text greater than 4096 tokens can be entered in main prompt and it must be correctly processed and memorized for the example to be used. from main chat?

**Comment by SilvaRaulEnrique** [comment]
in imartinez/privateGPT they are limit, is 4096 same here?
MODEL_N_CTX: Maximum token limit for the LLM model

**Comment by SilvaRaulEnrique** [comment]
@staviq : Exist another form for to utilizate any like privateGPT for read books and the response **fast** it's in another laguaje like **spanish**?, and use **CUDA**?

[privateGPT](https://github.com/imartinez/privateGPT)
